### PR TITLE
feat(tuples/map): use parallelized mapper

### PR DESCRIPTION
**Pull request by sno2**
```ts
type A = Call<
  Tuples.Map<Strings.Length>,
  ["a", "zvxjckvhjxhcv", "osduf", "aousdfo"]
>;
// type A = [1, 13, 5, 7]
```

### CODE REVIEW

File: src/internals/tuples/Tuples.ts [hunk 1]
```diff
@@ -159,9 +159,16 @@ export namespace Tuples {
...
+    return: this["arg1"] extends [...any[]]
+      ? MapFnImpl<this["arg1"], Extract<this["arg0"], Fn>>
+      : never;
   }
 
+  // This needs to be extracted out to show as a tuple type in editor tooling.
+  type MapFnImpl<a extends [...any[]], f extends Fn> = {
+    [K in keyof a]: Call<f, a[K]>;
+  };
+
```
Comment (ecyrbe) [inline src/internals/tuples/Tuples.ts:171]: Be carreful with naming. don't use single letters if possible. also `[..any[]]` should not be necessary.

Also i measured simple use cases and don't see perf improvements overall.

```suggestion
    return: MapFnImpl<this["arg1"], Extract<this["arg0"], Fn>>;
  }

  type MapFnImpl<tuple extends unknown[], fn extends Fn> = {
    [K in keyof tuple]: Call<fn, tuple[K]>;
  };
```
```

**Review by ecyrbe — COMMENTED** [review]
hello here is my review

File: src/internals/tuples/Tuples.ts (hunk 1 above)
Comment (sno2) [reply src/internals/tuples/Tuples.ts]: I have updated the code. Could've sworn that you needed `[...any[]]` to get it to work properly, but you're right it does work without it. Also, the benefits of this is that we are no longer limited by the recursion limit (10,000) and instead the union size limit which seems to be greater than 50,000.

**Comment by gvergnaud** [comment]
Nice, using a mapped types makes sense to me

**Pull request merged**